src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -182,6 +233,7 @@ private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind
             {
                 case BoundKind.PropertyGroup:
                     expr = BindIndexedPropertyAccess((BoundPropertyGroup)expr, mustHaveAllOptionalParameters: false, diagnostics: diagnostics);
+                    expr = BindIndexerDefaultArguments(expr, valueKind, diagnostics);
```

Comment:
>BindIndexerDefaultArguments [](start = 27, length = 27)

Does this have any effect?

Reply:
Yes. In some scenarios, we can call an indexer whose parameters are all optional by omitting the element access syntax entirely. Here's a test that demonstrates this:

https://github.com/dotnet/roslyn/blob/98cbb71e2c3030b42f37d813433f5bb57cc09c5e/src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs#L3080-L3139